cloudsim-plus/src/main/java/org/cloudbus/cloudsim/hosts/HostSimple.java

```diff
@@ -523,17 +521,35 @@ public final Host setActive(final boolean activate) {
             throw new IllegalStateException("The Host is failed and cannot be activated.");
         }
 
-        final boolean wasActive = this.active;
+        final double delay = activate ? powerModel.getStartupDelay() : powerModel.getShutDownDelay();
+        if (delay == 0) {
+           //If there is no delay, start up or shutdown the Host right away.
+           processHostActivation(activate);
+           return this;
+        }
+
+        final String msg = (activate ? "on" : "off") + " (expected time: {} seconds).";
+        final int tag = activate ? CloudSimTags.HOST_POWER_ON : CloudSimTags.HOST_POWER_OFF;
+        LOGGER.info("{}: {} is being powered " + msg, getSimulation().clockStr(), this, delay);
+        datacenter.schedule(delay, tag, this);
```

Comment:
When a delay is set, despite the host active attribute is only changed after such a delay,
the VmAllocationPolicy still places the VM immediately, no matter if the Host was activated or not.

Reply:
Currently, when the VmAllocationPolicy doesn't find an active Host to place a VM, it just gets an inactive one and calls setActive (which was activating the Host immediately). Now, if an active Host is not found, we must find an inactive suitable one, request its activation (calling setActive), and waiting for the broker to retry the VM creation later. This is already a broker feature. Check `DatacenterBroker.setRetryFailedVms()`.